arm_info:
  arm_base_joint:
    name: arm_base_joint
    type: Pulse
    dev_filename: /dev/uio4
    gpio_filename: /sys/class/gpio/gpio64/value
    to_meter_fraction: 200000.0 # should be in double type
    speed: 5000

  #shoulder_rotation_joint:
  #  name: shoulder_rotation_joint
  #  type: PWM
  #  dev_filename: /dev/uio7
  #  gpio_filename: /sys/class/gpio/gpio64/value
  #  duty_rates: [0.095, 0.093, 0.091, 0.089, 0.087, 0.085, 0.083, 
  #              0.081, 0.079, 0.077, 0.075, 0.073]
  #  angles:   [-1.5707963267948966, -1.2740903539558606, 
  #             -1.0471975511965976, -0.8377580409572781, 
  #             -0.6108652381980153, -0.3839724354387525, 
  #             -0.19198621771937624, 0.03490658503988659, 
  #             0.2617993877991494, 0.47123889803846897, 
  #             0.6806784082777886, 0.890117918517108]
  #  start_angle: -1.64060950 

  shoulder_flexion_joint:
    name: shoulder_flexion_joint
    type: PWM
    dev_filename: /dev/uio5
    gpio_filename: /sys/class/gpio/gpio64/value
    duty_rates: [0.052, 0.054, 0.056, 0.058, 0.06, 
                 0.062, 0.064, 0.066, 0.068, 0.07, 
                 0.072, 0.074, 0.076, 0.078, 0.08, 
                 0.082, 0.084, 0.086, 0.088, 0.09, 
                 0.093]

    angles: [-0.15533430342749532, -0.0942477796076938, 
             0.008726646259971648, 0.12740903539558604, 
             0.26354471705114374, 0.4118977034706618, 
             0.5550147021341968, 0.7051130178057091, 
             0.8534660042252271, 0.9738937226128359, 
             1.0838494654884787, 1.1641346110802178, 
             1.2374384396639797, 1.2950343049797925, 
             1.3421581947836396, 1.3788101090755203, 
             1.4119713648634127, 1.4381513036433275, 
             1.4538592669112764, 1.4660765716752369, 
             1.4817845349431857]

    start_angle: 1.48

  elbow_joint:
    name: elbow_joint
    type: PWM
    dev_filename: /dev/uio6
    gpio_filename: /sys/class/gpio/gpio64/value
    duty_rates: [0.078, 0.075, 0.073, 0.071, 0.069, 
                 0.067, 0.065, 0.063, 0.061, 
                 0.059, 0.057, 0.055, 0.053, 
                 0.051]

    angles: [0.28, 0.30543261909900765, 0.3892084231947355, 
             0.4974188368183839, 0.5602506898901798, 
             0.7853981633974483, 0.9546951008408984, 
             1.1205013797803596, 1.288052987971815, 
             1.4643312424232426, 1.616174887346749, 
             1.7976891295541593, 2.0001473227855016, 
             2.192133540504878]

    start_angle: 0.28


